### PR TITLE
feat(fastapi): major upgrade

### LINKED ISSUE #123: Fix auth tests in FastAPI template
Due to changes in version 3.3.0 of [FastAPI-Azure-Auth](https://github.com/Intility/fastapi-azure-auth), tokens generated in tests should be rewritten to use `jwk.construct` instead of `x5c` with base64 encoded signing certs.

See [diff](https://github.com/Intility/fastapi-azure-auth/compare/3.2.2...3.3.0#diff-9e08e8c7769db7b58089fd7659d75ca96df40bb7b5d50299d3a30abd54da9ad6) for context.

### LINKED ISSUE #125: Cleanup FastAPI template
* Optional auth
* Option to have a "less complete app"
* ~Arq support~

### PR DESCRIPTION
**Pull request by JonasKs**
- Remove `mypy` as an option, it is now always enabled
- Rework API structure to one file, making maintainability easier
- Make auth an option
  - Default: `FastAPI Azure AD (default)`
    - Do not test functionality already tested in FastAPI-Azure-Auth, and only test business logic for this specific app
  - Optional: `"Kong (no auth included)"`
- Rework `config.py` to be simpler
  - `Credentials` and `SettingsWithDerivedValues` instead of a bunch of different ones
- Remove all redis and cache support
  - Caching should mostly be done by Kong these days
- Bump versions of dependencies
- Change POSTGRES_HOST to POSTGRES_URL
  - Consistency change
- Option to create an app without example APIs (and tests) to reduce amount of files generated (for us that generates a lot of projects)

TODO:
- [x] Ensure docs are still correct 
  - [x] Auth
  - [x] Working of mypy
  - [x] Remove stuff about redis/cache
- [x] Ensure all options work
  - [x] Without SQLModel
  - [x] With SQLModel
  - [x] With Azure AD auth
  - [x] Without Azure AD auth
  - [x] Remove API files setting

@ingvaldlorentzen requests:

- [x] Sort on alphabet where possible
- [x] Remove system build settings from `pyproject.toml`
- [x] Add `asgi-correlation-id` by default 
- [x] Change Dockerfile structure
- [x] Remove `SECRET_KEY` - not in use 
  - Was implemented since I initially added encrypted cache support. This has been removed.
- [x] If no auth (Kong), only set CORS for development. 
- [x] Add `example.env`

Closes #125, closes #123

Since I'm force-pushing quite a lot, here's a few hashes for my own sanity:
- https://github.com/Intility/templates/commit/4fc6be20de05737ca683fea4a35c22481936c88e
- https://github.com/Intility/templates/commit/d606af3699b404393ed9b85cc98a961d029cb659
- https://github.com/Intility/templates/commit/0026262aa16168229fbc439ff4a1828e2fb86eb4

### CODE REVIEW

File: fastapi/README.md [hunk 1]
```diff
@@ -12,5 +12,8 @@ based on selections.
 
 
 ## Usage
-* `cruft create . --directory="fastapi"` to use from source
+* `poetry init` from `templates` folder.
```
Comment (h3rmanj) [inline fastapi/README.md:15]: Should you always clone templates repo first?

File: fastapi/README.md [hunk 2]
```diff
@@ -12,5 +12,8 @@ based on selections.
...
+* `poetry add cruft`
+* `cruft create . --directory="fastapi"` to use from source (current commit)
+  * `cookiecutter fastapi` to use from current state (used while testing uncommitted changes)
 * `cruft create https://github.com/Intility/templates.git --directory="fastapi"` to use from GitHub
```
Comment (h3rmanj) [inline fastapi/README.md:19]: are all these steps necessary to init directly from github? cloning the repo first and all?

**Review by h3rmanj — COMMENTED** [review]
LGTM with a couple of nitpicks

File: fastapi/README.md (hunk 2 above)
Comment (JonasKs) [reply fastapi/README.md:19]: These steps are for when editing source (like I'm doing right now). 

I'll make it more clear.😊

**Comment by JonasKs** [comment]
Thank you @h3rmanj! I still need to rework the docker files and the README, but a bunch of new things has been added. I'll finish early next week.

**Review by h3rmanj — APPROVED** [review]
LGTM :100:

**Comment by ingvaldlorentzen** [comment]
👍

**Pull request merged**